source/public/scripts/main.journal.js

```diff
@@ -0,0 +1,127 @@
+document.addEventListener("DOMContentLoaded", function () {
+    // Select the modal and various buttons
+    var modal = document.getElementById("task-modal");
+    var btn = document.querySelector(".new-task");
+    var saveBtn = document.querySelector(".save-task");
+    var cancelBtn = document.querySelector(".cancel-task");
+    var deleteBtnModal = document.querySelector(".delete-task");
+    var taskContainer = document.getElementById("task-container");
+
+    // Initialize editingIndex to -1 to indicate no task is being edited
+    let editingIndex = -1;
+
+    // Show the modal when the "New Task" button is clicked
+    btn.onclick = function () {
+      editingIndex = -1; // Reset editing index when adding a new task
+      modal.style.display = "block"; // Show the modal
+      clearModalFields(); // Clear any existing data in the modal fields
+    }
+
+    // Hide the modal if the user clicks outside of it
+    window.onclick = function (event) {
+      if (event.target == modal) {
+        modal.style.display = "none";
+      }
+    }
+
+    // Save the task when the "Save" button is clicked
+    saveBtn.onclick = function () {
+      // Get task details from modal input fields
+      var taskDesc = document.getElementById("task-desc").value;
+      var taskType = document.getElementById("task-type").value;
+      var taskProject = document.getElementById("task-project").value;
+
+      // Retrieve existing tasks from local storage or initialize as empty array
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
```

Comment:
This works right now since we don't have infrastructure in place for entries for multiple days but would change in the future where tasks are inside an entry object (according to the schema we came up with). If you could update this to reflect that schema I think it would be helpful as we continue working on the functionalities